variant-remapping-automation/prepare_remapping_table.py

```diff
@@ -0,0 +1,320 @@
+        return 'DBSNP, EVA'
+    elif assembly in dbsnp_assemblies:
+        return 'DBSNP'
+    elif assembly in eva_assemblies:
+        return 'EVA'
+
+
+def release_version_exists(private_config_xml_file, release_version):
+    with get_metadata_connection_handle('production_processing', private_config_xml_file) as pg_conn:
+        query = f"select * from eva_progress_tracker.remapping_tracker where release_version={release_version}"
+        return get_all_results_for_query(pg_conn, query)
+
+
+def get_asm_no_ss_ids(tax_asm_studies, noah_prj_dir, codon_prj_dir):
+    """
+    For each taxonomy and assembly, get the total number of ssids to remap, by reading the accessioning report
+    for all the studies which needs to be remapped
+    """
+    tax_asm_ssids = defaultdict(lambda: defaultdict(int))
+    for tax, asm_studies in tax_asm_studies.items():
+        for asm, studies in asm_studies.items():
+            no_of_ss_ids_in_asm = 0
+            for study in studies:
+                # get path to accessioned file on both noah and codon
+                noah_file_path, codon_file_path = get_accession_reports_for_study(study, noah_prj_dir, codon_prj_dir)
+                for file in noah_file_path:
+                    no_of_ss_ids_in_asm += get_no_of_ss_ids_in_file(file)
+                for file in codon_file_path:
+                    no_of_ss_ids_in_asm += get_no_of_ss_ids_in_file(file)
+
+        logger.info(f"No of ss ids in taxonomy {tax} assembly {asm} are {no_of_ss_ids_in_asm}")
+        tax_asm_ssids[tax][asm] = no_of_ss_ids_in_asm
+
+    return tax_asm_ssids
+
+
+def get_no_of_ss_ids_in_file(path):
+    """
+    Read a file to figure out the number of ss ids in that file
+    skip line start with #
+    read line where 3rd entry starts with ss (assuming all files have same sequence of headers and ss id is always 3rd)
+    """
+    no_of_ss_ids_in_file = 0
+    with gzip.open(path, 'rt') as f:
+        for line in f:
+            if line.startswith('#'):
+                continue
+            elif line.split("\t")[2].startswith("ss"):
+                no_of_ss_ids_in_file = no_of_ss_ids_in_file + 1
+
+    print(f"No of ss ids in file {path} : {no_of_ss_ids_in_file}")
+    return no_of_ss_ids_in_file
+
+
+def get_accession_reports_for_study(study, noah_file_path, codon_file_path):
+    """
+    Given a study, find the accessioning report path for that study on both noah and codon
+    look for a file ending with accessioned.vcf.gz (assuming only one file with the given pattern will be present)
+    if more than one files are present, take the first one
+    """
+
+    noah_file = glob.glob(f"{noah_file_path}/{study}/60_eva_public/*accessioned.vcf.gz")
+    codon_file = glob.glob(f"{codon_file_path}/{study}/60_eva_public/*accessioned.vcf.gz")
+
+    if not noah_file and not codon_file:
+        logger.error(f"Could not find any file in Noah or Codon for Study {study}")
```

Comment:
I am wondering why we don't just exit at this point?

Reply:
because we are running for all the assemblies together, I thought it would be better to log this error and continue, instead of failing.